service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java

```diff
@@ -17,26 +17,29 @@
 
 import static java.util.Objects.isNull;
 import static org.apache.commons.lang.StringUtils.capitalize;
+import static org.apache.commons.lang.StringUtils.uncapitalize;
 import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
 import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.getFirstRespondentLastName;
 
-@SuppressWarnings({"LineLength", "VariableDeclarationUsageDistance"})
 public abstract class AbstractEmailContentProvider {
 
     final String uiBaseUrl;
-    final DateFormatterService dateFormatterService;
-    final HearingBookingService hearingBookingService;
+    private final DateFormatterService dateFormatterService;
+    private final HearingBookingService hearingBookingService;
 
-    protected AbstractEmailContentProvider(String uiBaseUrl, DateFormatterService dateFormatterService,
+    protected AbstractEmailContentProvider(String uiBaseUrl,
+                                           DateFormatterService dateFormatterService,
                                            HearingBookingService hearingBookingService) {
         this.uiBaseUrl = uiBaseUrl;
         this.dateFormatterService = dateFormatterService;
         this.hearingBookingService = hearingBookingService;
     }
 
     @SuppressWarnings("unchecked")
-    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(CaseDetails caseDetails) {
-        List<String> ordersAndDirections = buildOrdersAndDirections((Map<String, Object>) caseDetails.getData().get("orders"));
+    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(CaseDetails caseDetails, CaseData caseData) {
```

Comment:
Passing same thing as two different arguments (in slightly different format - Map vs CaseData) does not make much sense. Somebody who will have to use `getCasePersonalisationBuilder` in the future will have bad headache.

Reply:
I think I could stick with just using caseDetails, I think I would have to do something like:
`        List<Map<String, Object>> respondents = (List<Map<String, Object>>) data.get("respondents1");
        Map<String, Object> value = (Map<String, Object>) respondents.get(0).get("value");
        Map<String, Object> party = (Map<String, Object>) value.get("party");
        Map<String, Object> lastName = (Map<String, Object>) value.get("lastName");`
to get the respondent last name. Bit of a headache, I don't like this way of extracting information; using caseData is a lot cleaner, but did not want to change existing code too much so added caseData as argument for simplicity. If preferred I could try retrieving lastName using Map format

Reply:
I'd personally, as long as time allows, rewrite method to use CaseData. It will be, as you said, way more readable than current implementation. These operations on maps are tech debt ash should be killed sooner than later.